curl/src/client.cc

```diff
@@ -296,63 +314,54 @@ namespace leatherman { namespace curl {
 
     void client::set_timeouts(context& ctx)
     {
-        auto result = curl_easy_setopt(_handle, CURLOPT_CONNECTTIMEOUT_MS, ctx.req.connection_timeout());
-        if (result != CURLE_OK) {
-            throw http_request_exception(ctx.req, curl_easy_strerror(result));
-        }
-        result = curl_easy_setopt(_handle, CURLOPT_TIMEOUT_MS, ctx.req.timeout());
-        if (result != CURLE_OK) {
-            throw http_request_exception(ctx.req, curl_easy_strerror(result));
-        }
+        curl_easy_setopt_maybe<long>(ctx, CURLOPT_CONNECTTIMEOUT_MS, ctx.req.connection_timeout());
+        curl_easy_setopt_maybe<long>(ctx, CURLOPT_TIMEOUT_MS, ctx.req.timeout());
     }
 
     void client::set_write_callbacks(context& ctx)
     {
-        auto result = curl_easy_setopt(_handle, CURLOPT_HEADERFUNCTION, write_header);
-        if (result != CURLE_OK) {
-            throw http_request_exception(ctx.req, curl_easy_strerror(result));
-        }
-        result = curl_easy_setopt(_handle, CURLOPT_HEADERDATA, &ctx);
-        if (result != CURLE_OK) {
-            throw http_request_exception(ctx.req, curl_easy_strerror(result));
-        }
-        result = curl_easy_setopt(_handle, CURLOPT_WRITEFUNCTION, write_body);
-        if (result != CURLE_OK) {
-            throw http_request_exception(ctx.req, curl_easy_strerror(result));
-        }
-        result = curl_easy_setopt(_handle, CURLOPT_WRITEDATA, &ctx);
-        if (result != CURLE_OK) {
-            throw http_request_exception(ctx.req, curl_easy_strerror(result));
-        }
+        curl_easy_setopt_maybe<curl_header_callback>(ctx, CURLOPT_HEADERFUNCTION, write_header);
+        curl_easy_setopt_maybe<void*>(ctx, CURLOPT_HEADERDATA, &ctx);
+        curl_easy_setopt_maybe<curl_write_callback>(ctx, CURLOPT_WRITEFUNCTION, write_body);
+        curl_easy_setopt_maybe<void*>(ctx, CURLOPT_WRITEDATA, &ctx);
     }
 
     void client::set_ca_info(context& ctx){
-        if (_ca_cert != ""){
-            auto result = curl_easy_setopt(_handle, CURLOPT_CAINFO, _ca_cert.c_str());
-            if (result != CURLE_OK) {
-                throw http_request_exception(ctx.req, curl_easy_strerror(result));
-            }
+        if (_ca_cert == "") {
+            return;
         }
+        
+        curl_easy_setopt_maybe<const char*>(ctx, CURLOPT_CAINFO, _ca_cert.c_str());
+    }
+
+    void client::set_ca_info(std::string const& url, std::string const& file_path){
```

Comment:
Oh, it's a helper for the exception that will be thrown. That's somewhat confusing. I'd be ok with set_ca_info just throwing one type of exception.